flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/rules/logical/PushPartitionIntoTableSourceScanRule.java

```diff
@@ -176,6 +177,14 @@ public void onMatch(RelOptRuleCall call) {
                         defaultPruner,
                         allPredicates._1(),
                         inputFieldNames);
+
+        // If remaining partitions are empty, it means that there are no partitions are selected
+        // after partition prune. We can directly optimize the RelNode to Empty FlinkLogicalValues.
+        if (remainingPartitions.isEmpty()) {
+            RelNode emptyValue = relBuilder.push(filter).empty().build();
```

Comment:
Can it be `relBuilder.empty().build()`?

Reply:
> Can it be `relBuilder.empty().build()`?

It must need a leaf node, otherwise the `NoSuchElementException` will be thrown.